pcsx2/CDVD/CsoFileReader.cpp

```diff
@@ -256,18 +264,34 @@ int CsoFileReader::ReadChunk(void* dst, s64 chunkID)
 		// This might be less bytes than frameRawSize in case of padding on the last frame.
 		// This is because the index positions must be aligned.
 		const u32 readRawBytes = fread(m_readBuffer.get(), 1, frameRawSize, m_src);
+		bool success = false;
 
-		m_z_stream->next_in = m_readBuffer.get();
-		m_z_stream->avail_in = readRawBytes;
-		m_z_stream->next_out = static_cast<Bytef*>(dst);
-		m_z_stream->avail_out = m_frameSize;
+		if (m_uselz4)
+		{
+			int src_size = static_cast<int>(readRawBytes);
```

Comment:
All of these should be const.

Reply:
They can't, LZ4 does not expect them to be const, making them const will issue an error (C++ is too nickpicky with data types).

This is the header of the LZ4 function used:

int LZ4_decompress_safe_partial(const char* src, char* dst, int compressedSize, int targetOutputSize, int dstCapacity)

As you can see, only the first parameter is const, the rest aren't. If I make them const, the compiler will complain.

Reply:
The value is copied when passing it to a parameter, so the constness of the variable it's coming from is irrelevant. It's not passed by reference.